code/game/machinery/doors/door.dm

```diff
@@ -178,6 +178,14 @@
 			else
 				do_animate("deny")
 		return
+	if(istype(AM, /obj/structure/janitorialcart))

```

Comment:
Instead of doing this. Just make the if-check above this be `if(istype(AM, /obj/structure/bed/chair/wheelchair) || istype(AM, /obj/structure/janitorialcart))`

And change line 174 to be `var/obj/structure/struct = AM`

Reply:
I'd like to but the pulling var is defined on the janitorialcart and wheelchair respectively.